address/address.go

```diff
@@ -129,3 +132,77 @@ func ToBech32(bc *Bech32) (string, error) {
 
 	return bech, nil
 }
+
+// FromBlech32 decodes a blech32 encoded string, returning the human-readable
+// part and the data part excluding the checksum.
+func FromBlech32(address string) (*Blech32, error) {
+	// Decode the bech32 encoded address.
+	_, data, err := blech32.Decode(address)
+	if err != nil {
+		return nil, err
+	}
+
+	// The first byte of the decoded address is the witness version, it must
+	// exist.
+	if len(data) < 1 {
+		return nil, fmt.Errorf("no witness version")
+	}
+
+	// ...and be <= 16.
+	version := data[0]
+	if version > 16 {
+		return nil, fmt.Errorf("invalid witness version: %v", version)
+	}
+
+	// The remaining characters of the address returned are grouped into
+	// words of 5 bits. In order to restore the original witness program
+	// bytes, we'll need to regroup into 8 bit words.
+	regrouped, err := blech32.ConvertBits(data[1:], 5, 8, false)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(regrouped) < 2 || len(regrouped) > 40+33 {
+		return nil, fmt.Errorf("invalid data length")
+	}
+
+	// For witness version 0, address MUST be exactly 20+33 or 32+33 bytes.
+	if version == 0 && len(regrouped) != 20+33 && len(regrouped) != 32+33 {
+		return nil, fmt.Errorf("invalid data length for witness "+
+			"version 0: %v", len(regrouped))
+	}
+
+	return &Blech32{version, nil, regrouped}, nil
+}
+
+// ToBlech32 encodes a byte slice into a blech32 string
+func ToBlech32(hrp string, witnessVersion byte, witnessProgram []byte) (string, error) {
```

Comment:
similarly to the `ToBech32` function, we should expect a pointer to Blech32 type:
```suggestion
func ToBlech32(bl *Blech32) (string, error) {
```